packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/fragment.js

```diff
@@ -155,8 +154,16 @@ function is_static_element(node) {
 			return false;
 		}
 
-		if (node.name.includes('-')) {
-			return false; // we're setting all attributes on custom elements through properties
+		if (
+			attribute.value !== true &&
+			!is_text_attribute(attribute) &&
+			// if the attribute is not a text attribute but is inlinable we will directly inline it in the
+			// the template so if it's not inlinable we can return false but if it is inlinable (or a text)
+			// attribute we keep count of the inlinable attributes so that if all the attributes are inlinable
+			// we deem the node as static
```

Comment:
this would be easier to read if it's not all one long run-on sentence :laughing:

> If it is an inlinable (or text) attribute we keep count of the inlinable attributes and if all the attributes are inlinable then we deem the node as static

where does this happen? are you referring to what occurs within `is_inlinable_expression`? if so, I think we could remove " we keep count of the inlinable attributes" as it's not really counting them, but just checking them all. or were you referring to something else?

```suggestion
			// If the attribute is not a text attribute but is inlinable we will directly inline it in the
			// the template. If it's not inlinable we can return false. If it is an inlinable (or text)
			// attribute we keep count of the inlinable attributes and if all the attributes are inlinable
			// then we deem the node as static
```